src/reconcile/rec_write.c

```diff
@@ -1395,7 +1416,7 @@ __rec_txn_read(WT_SESSION_IMPL *session, WT_RECONCILE *r,
 			 * discard an uncommitted update.
 			 */
 			if (F_ISSET(r, WT_REC_UPDATE_RESTORE) &&
-			    *updp != NULL && uncommitted) {
+			    *updp != NULL && (uncommitted || prepared)) {
```

Comment:
Will the evaluation of (uncommitted || prepared) can ever be true ? as we continue in the loop at line 1373, if it is "true".

Reply:
It can be true, we discussed this offline